### PR TITLE
ftx: query kline using rest api

### PR DESCRIPTION
**Pull request by ycdesu**
FTX only supports REST Kline api

### CODE REVIEW

File: pkg/exchange/ftx/convert.go [hunk 1]
```diff
@@ -129,3 +129,19 @@ func toGlobalTrade(f fill) (types.Trade, error) {
 		IsIsolated:    false,
 	}, nil
 }
+
+func toGlobalKLine(symbol string, interval types.Interval, h Candle) (types.KLine, error) {
+	return types.KLine{
+		Exchange:  types.ExchangeFTX.String(),
+		Symbol:    symbol,
```
Comment (c9s) [inline pkg/exchange/ftx/convert.go:136]: ToGlobalSymbol? because these klines will be stored in the db

File: pkg/exchange/ftx/exchange.go [hunk 2]
```diff
@@ -142,7 +142,51 @@ func (e *Exchange) QueryAccountBalances(ctx context.Context) (types.BalanceMap,
...
+		if err != nil {
+			return nil, err
+		}
+		kline = append(kline, globalKline)
+	}
+	return kline, nil
+}
+
+func isIntervalSupportedInKLine(interval types.Interval) bool {
+	_, ok := map[int]struct{}{
```
Comment (c9s) [inline pkg/exchange/ftx/exchange.go:180]: pull out the predefined map?

File: pkg/exchange/ftx/exchange.go [hunk 3]
```diff
@@ -142,7 +142,51 @@ func (e *Exchange) QueryAccountBalances(ctx context.Context) (types.BalanceMap,
...
+		15:    {},
```
Comment (c9s) [inline pkg/exchange/ftx/exchange.go:181]: we need at least 1m and 5m for back testing

**Review by c9s — APPROVED** [review]

File: pkg/exchange/ftx/convert.go (hunk 1 above)
Comment (ycdesu) [reply pkg/exchange/ftx/convert.go:136]: How to convert to global symbol, such as btc-perp?

File: pkg/exchange/ftx/exchange.go (hunk 3 above)
Comment (ycdesu) [reply pkg/exchange/ftx/exchange.go:181]: Yes it's supported by ftx.

File: pkg/exchange/ftx/convert.go (hunk 1 above)
Comment (c9s) [reply pkg/exchange/ftx/convert.go:136]: perp is for future, we don't support futures for now. will need the design for futures

Comment (c9s) [reply pkg/exchange/ftx/convert.go:136]: simply convert BTC/USDT to BTCUSDT ?

Comment (ycdesu) [reply pkg/exchange/ftx/convert.go]: done

**Review by c9s — APPROVED** [review]

**Pull request merged**